test/utils/mocha.ts

```diff
@@ -283,14 +283,46 @@ export function expectRule(id: string, ruleset?: string): DataLayerRule {
   return rule!;
 }
 
+/**
+ * Valid FullStory object method names.
+ */
+type FSMethodName = 'event' | 'identify' | 'log' | 'setVars' | 'setUserVars';
+
 /**
  * A convenience method for `expectParams` that checks the global FullStory object.
  * @param methodName FullStory API function (`methodName` arg for `expectParams`)
  * @param namespace Global object if FullStory is not `FS`
  */
-export function expectFS(methodName: 'event' | 'identify' | 'log' | 'setVars' | 'setUserVars',
-  namespace = 'FS'): any[] {
+export function expectFS(methodName: FSMethodName, namespace = 'FS'): any[] {
   const fs = expectGlobal(namespace);
   expect(fs).to.be.ok;
   return expectParams(fs, methodName);
 }
+
+/**
+ * Similar to `expectFS`, `waitForFS` is a convenicne method for `expectParams` that
+ * checks the global FullStory object. Unlike `expectFS`, `waitForFS` supports waiting
+ * for a global FullStory object method call within a timeout period.
+ * @param methodName FullStory API function (`methodName` arg for `expectParams`)
+ * @param namespace Global object if FullStory is not `FS`
+ * @param timeout Time in milliseconds after which the Promise is rejected
+ */
+export function waitForFS(methodName: FSMethodName, namespace = 'FS', timeout: number = 1000): Promise<any[]> {
```

Comment:
I added this convenience function to support testing the edge case this PR intends to resolve. To verify this behavior, I needed to change data layer values and wait for event dispatching to carry through calls to rule destinations. This can take a variable duration to propagate, so I've checked the condition every 50ms here up until the configured timeout. Interested if you all have better ideas for how to approach this.

Reply:
I think the challenge we'd run into is there are multiple waits within the test -- wait for first rule destination call, wait for second rule destination call, and wait for a defined period of time before ensuring no further rule destinations were called. I don't think we'd be able to call `done()` until the last condition were true, whereas it feels like async/await better handles the stepped nature of the waits. Let me know if any of that doesn't make sense. I might not be fully understanding how we could thread `done` through the wait conditions in this test.